### PR TITLE
Ess gui 1554 slicer

### PR DESCRIPTION
**Pull request by murphyryanp**
This is mostly a "merge rescue operation" from @rozyczko branch 'ESS_GUI_1381_slicer'. I pulled in all non-conflicting changes from the 'ESS_GUI_1381_slicer' branch into the 'ESS_GUI' branch, and then removed some features that I think require further development. Key changes regarding issue #1554 are summarized below:

- Fixes the critical issue regarding the annulus slicer not opening. Also, the parameters can be set and are updated in the SlicerParameters window.
- Removed the batch slicing tab in the SlicerParameter widget and the associated widgets (which are commented out) since it seems this needs more tweaking. However, the batch slicing is a really nice feature (that I would like to use!) but the auto-saving and fitting selection features are currently inactive. Should we save the batch slicing feature for a later release?
- Minimum limits to the annulus slicer changed from 0.01 to dqmin, where dqmin is defined by the input 2D data. This is for VSANS or other low-q 2D data that may require annulus slices with widths less than 0.01 A^-1.
- Maximum limits to the annulus slicer changed to the maximum of |qx| (instead of the minimum), which is mainly an issue for offset 2D data. If the 2D data has an offset, then the user won't be limited to setting the annulus to the edge closest to the offset center. Instead, the limit becomes the edge farthest from the offset center. (We may also want to consider adding an evaluation of ymax to handle the possible cases where data is offset in the qy direction instead of the qx direction.)

### CODE REVIEW

**Comment by murphyryanp** [comment]
So should I merge this Pull Request as is, but also create a new issue for Paul's suggestion to integrate context manager for all the slicers?

**Comment by pkienzle** [comment]
I turned my comment into a separate issue.  Still needs verification that it functions as expected on windows before merging, unless @rozyczko has already done so?

**Comment by butlerpd** [comment]
5.0 ready for testing on Win

File: src/sas/qtgui/Plotting/SlicerParameters.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def __init__(self, parent=None,
         self.active_plots = active_plots
 
         # Initially, Apply is disabled
-        self.cmdApply.setEnabled(False)
+        #self.cmdApply.setEnabled(False)
```
Comment (murphyryanp) [inline src/sas/qtgui/Plotting/SlicerParameters.py:44]: @butlerpd Here's one example of the command commented out related to the batch slicer tab. The batch slicing tab was removed in the .ui file.

**Comment by butlerpd** [comment]
LOL  missed that.  but this should be merged I think.

Eventually, depending on whether @rozyczko thinks the batch slicing should be restarted from master or continued from the previous branch he may want to merge ESS_GUI master back into his batch slicer working branch I guess.

**Comment by rozyczko** [comment]
I'll most likely reapply changes from my old branch to ESS_GUI and finish off the slicer. Soon.

**Pull request merged**